### PR TITLE
fix: container executor negative test

### PR DESCRIPTION
**Pull request by nicufk**
## Pull request description 
Closes https://github.com/kubeshop/testkube/issues/3109

## Checklist (choose whats happened)

- [ ] breaking change! (describe)
- [ ] tested locally
- [ ] tested on cluster
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

## Breaking changes

-

## Changes

-

## Fixes

-

### CODE REVIEW

**Review by vLia — APPROVED** [review]

**Comment by windowsrefund** [comment]
@nicufk Thank you for this!

File: pkg/executor/containerexecutor/containerexecutor.go [hunk 1]
```diff
@@ -492,10 +493,33 @@ func (c *ContainerExecutor) updateResultsFromPod(
...
+			result.Status = testkube.ExecutionStatusPassed
+			result.Output = result.Output + "\nTest run was expected to fail, and it failed as expected"
+		} else {
+			c.log.Debugw("test run was expected to fail - the result will be reversed", "test", execution.TestName)
+			execution.ExecutionResult.Status = testkube.ExecutionStatusFailed
+			result.Status = testkube.ExecutionStatusFailed
+			result.Output = result.Output + "\nTest run was expected to fail, the result will be reversed"
+		}
+	}
+	err := c.repository.UpdateResult(ctx, execution.Id, *execution)
```
Comment (vsukhin) [inline pkg/executor/containerexecutor/containerexecutor.go:517]: should we move it to if above?

**Review by vsukhin — COMMENTED** [review]
one comment

File: pkg/executor/containerexecutor/containerexecutor.go (hunk 1 above)
Comment (nicufk) [reply pkg/executor/containerexecutor/containerexecutor.go]: Done

**Review by vsukhin — APPROVED** [review]

**Pull request merged**